lib/rules/sort-flags.ts

```diff
@@ -0,0 +1,181 @@
+import type { Literal } from "estree"
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+
+export default createRule("sort-flags", {
+    meta: {
+        docs: {
+            description: "require the regex flags to be sorted",
+            // TODO Switch to recommended in the major version.
+            // recommended: true,
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    order: {
```

Comment:
I think we should not offer this option. For two reasons:

1. We should enforce the order that the [ECMAScript spec](https://tc39.es/ecma262/#sec-get-regexp.prototype.flags) requires for the `RegExp.properties.flags` property.
2. A custom order might interfere with code printers/formatters (like Prettier) that automatically enforce `RegExp.properties.flags` order.

I actually also had this option in the first version of `clean-regex/sort-flags` but removed it for the above reasons.